pkg/local_object_storage/metabase/metadata.go

```diff
@@ -0,0 +1,742 @@
+)
+
+const (
+	intValLen      = 33                              // prefix byte for sign + fixed256 in metaPrefixAttrIDInt
+	attrIDFixedLen = 1 + oid.Size + utf8DelimiterLen // prefix first
+)
+
+var (
+	maxUint256 = new(big.Int).SetBytes([]byte{255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255,
+		255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255})
+	maxUint256Neg = new(big.Int).Neg(maxUint256)
+)
+
+var errInvalidCursor = errors.New("invalid cursor")
+
+func invalidMetaBucketKeyErr(key []byte, cause error) error {
+	return fmt.Errorf("invalid meta bucket key (prefix 0x%X): %w", key[0], cause)
+}
+
+// TODO: fill on migration.
+// TODO: ROOT and PHY props.
+// TODO: cleaning on obj removal.
+func putMetadata(tx *bbolt.Tx, cnr cid.ID, id oid.ID, ver version.Version, owner user.ID, typ object.Type, creationEpoch uint64,
+	payloadLen uint64, pldHash, pldHmmHash, splitID []byte, parentID, firstID oid.ID, attrs []object.Attribute) error {
+	metaBkt, err := tx.CreateBucketIfNotExists(metaBucketKey(cnr))
+	if err != nil {
+		return fmt.Errorf("create meta bucket for container: %w", err)
+	}
+	idk := [1 + oid.Size]byte{metaPrefixID}
+	copy(idk[1:], id[:])
+	if err := metaBkt.Put(idk[:], nil); err != nil {
+		return fmt.Errorf("put object ID to container's meta bucket: %w", err)
+	}
+
+	var keyBuf keyBuffer
+	if err = putPlainAttribute(metaBkt, &keyBuf, id, object.FilterVersion, ver.String()); err != nil {
+		return err
+	}
+	if err = putPlainAttribute(metaBkt, &keyBuf, id, object.FilterOwnerID, string(owner[:])); err != nil {
+		return err
+	}
+	if err = putPlainAttribute(metaBkt, &keyBuf, id, object.FilterType, typ.String()); err != nil {
+		return err
+	}
+	if err = putIntAttribute(metaBkt, &keyBuf, id, object.FilterCreationEpoch, new(big.Int).SetUint64(creationEpoch)); err != nil {
+		return err
+	}
+	if err = putIntAttribute(metaBkt, &keyBuf, id, object.FilterPayloadSize, new(big.Int).SetUint64(payloadLen)); err != nil {
+		return err
+	}
+	if err = putPlainAttribute(metaBkt, &keyBuf, id, object.FilterPayloadChecksum, string(pldHash)); err != nil {
+		return err
+	}
+	if err = putPlainAttribute(metaBkt, &keyBuf, id, object.FilterPayloadHomomorphicHash, string(pldHmmHash)); err != nil {
+		return err
+	}
+	if len(splitID) > 0 {
+		// TODO: deprecated, maybe not index?
+		if err = putPlainAttribute(metaBkt, &keyBuf, id, object.FilterSplitID, string(splitID)); err != nil {
+			return err
+		}
+	}
+	if !firstID.IsZero() {
+		if err = putPlainAttribute(metaBkt, &keyBuf, id, object.FilterFirstSplitObject, string(firstID[:])); err != nil {
+			return err
+		}
+	}
+	if !parentID.IsZero() {
+		if err = putPlainAttribute(metaBkt, &keyBuf, id, object.FilterParentID, string(parentID[:])); err != nil {
+			return err
+		}
+	}
+	for i := range attrs {
+		ak, av := attrs[i].Key(), attrs[i].Value()
+		if n, isInt := parseInt(av); isInt && n.Cmp(maxUint256Neg) >= 0 && n.Cmp(maxUint256) <= 0 {
```

Comment:
so we just do not index ints that overflows us? isn't it an error? should we support such cases?

Reply:
yes, this is OK, fixed256 is an enormous value covering any usecase nowadays